tests/test_filtsmooth/test_gaussfiltsmooth/test_kalmanposterior.py

```diff
@@ -1,156 +1,185 @@
+
+
+def test_locations(posterior, problem):
+    """Locations are stored correctly."""
+    *_, obs, times, states = problem
+    np.testing.assert_allclose(posterior.locations, np.sort(posterior.locations))
+    np.testing.assert_allclose(posterior.locations, times)
+
+
+def test_getitem(posterior):
+    """Getitem performs as expected."""
+
+    np.testing.assert_allclose(posterior[0].mean, posterior.state_rvs[0].mean)
+    np.testing.assert_allclose(posterior[0].cov, posterior.state_rvs[0].cov)
+
+    np.testing.assert_allclose(posterior[-1].mean, posterior.state_rvs[-1].mean)
+    np.testing.assert_allclose(posterior[-1].cov, posterior.state_rvs[-1].cov)
+
+    np.testing.assert_allclose(posterior[:].mean, posterior.state_rvs[:].mean)
+    np.testing.assert_allclose(posterior[:].cov, posterior.state_rvs[:].cov)
+
+
+def test_state_rvs(posterior):
+    """RVs are stored correctly."""
+
+    assert isinstance(posterior.state_rvs, _RandomVariableList)
+    assert len(posterior.state_rvs[0].shape) == 1
+
+
+def test_call_error_if_small(posterior):
+    """Evaluating in the past of the data raises an error."""
+    assert -0.5 < posterior.locations[0]
+    with pytest.raises(ValueError):
+        posterior(-0.5)
+
+
+def test_call_vectorisation(posterior):
+    """Evaluation allows vector inputs."""
+    locs = np.arange(0, 1, 20)
+    evals = posterior(locs)
+    assert len(evals) == len(locs)
+
+
+def test_call_interpolation(posterior):
+    """Interpolation is possible and returns a Normal RV."""
+    assert posterior.locations[0] < 9.88 < posterior.locations[-1]
+    assert 9.88 not in posterior.locations
+    out_rv = posterior(9.88)
+    assert isinstance(out_rv, pnrv.Normal)
+
+
+def test_call_to_discrete(posterior):
+    """Called at a grid point, the respective disrete solution is returned."""
+
+    first_point = posterior.locations[0]
+    np.testing.assert_allclose(posterior(first_point).mean, posterior[0].mean)
+    np.testing.assert_allclose(posterior(first_point).cov, posterior[0].cov)
+
+    final_point = posterior.locations[-1]
+    np.testing.assert_allclose(posterior(final_point).mean, posterior[-1].mean)
+    np.testing.assert_allclose(posterior(final_point).cov, posterior[-1].cov)
+
+    mid_point = posterior.locations[4]
+    np.testing.assert_allclose(posterior(mid_point).mean, posterior[4].mean)
+    np.testing.assert_allclose(posterior(mid_point).cov, posterior[4].cov)
+
+
+def test_call_extrapolation(posterior):
+    """Extrapolation is possible and returns a Normal RV."""
+    assert posterior.locations[-1] < 30.0
+    out_rv = posterior(30.0)
+    assert isinstance(out_rv, pnrv.Normal)
+
+
+@pytest.fixture
+def size():
+    return 10
+
+
+@pytest.fixture
+def size():
+    return (12,)
+
+
+@pytest.fixture
+def locs(posterior):
+    return None
+
+
+@pytest.fixture
+def locs(posterior):
+    return posterior.locations[[2, 3]]
+
+
+@pytest.fixture
+def locs():
+    return np.arange(0, 0.5, 0.025)
```

Comment:
I'm wondering: Is this also called in the tests? I'm still a bit confused with pytest, and I'm surprised to see `locs` with different function signatures, so I just want to make sure things work as I'd expect them to.

Reply:
I think so... I see that this looks weird. Shall I put the `posterior` argument into the other fixtures, as well?

Reply:
Maybe in this one (`locs`) to make things a bit more clear? Though as long as all of them run correctly, maybe it's also fine as it is.

Reply:
Oh but I just saw in line 141 that there is a `locs(posterior)` that does not use the `posterior` in the function, so that should be removed (if we keep this function as it is, too).

Reply:
I put it in, for the sake of a uniform function signature. I dont think it matters too much, though if this turns out to be bad, that will be an easy fix.